indico/modules/categories/blueprint.py

```diff
@@ -16,9 +16,46 @@
 
 from __future__ import unicode_literals
 
-from indico.modules.categories.controllers import RHCategoryStatistics
+from indico.modules.categories.controllers.display import RHCategoryDisplay, RHCategoryStatistics
+from indico.modules.categories.controllers.management import RHCategoryManage
+from indico.web.flask.util import redirect_view, make_compat_redirect_func
 from indico.web.flask.wrappers import IndicoBlueprint
 
-_bp = IndicoBlueprint('categories', __name__, template_folder='templates', url_prefix='/category')
+from MaKaC.webinterface.rh import calendar, categoryDisplay
 
-_bp.add_url_rule('/<categId>/statistics', 'category-statistics', RHCategoryStatistics)
+_legacy_bp = IndicoBlueprint('category', __name__, template_folder='templates', virtual_template_folder='categories',
+                             url_prefix='/category/<int:categId>')
+
+# Short URLs
+_legacy_bp.add_url_rule('!/categ/<categId>', view_func=redirect_view('.display'), strict_slashes=False)
+_legacy_bp.add_url_rule('!/c/<categId>', view_func=redirect_view('.display'), strict_slashes=False)
+
+# Display
+_legacy_bp.add_url_rule('/', 'categoryDisplay', categoryDisplay.RHCategoryDisplay)
+_legacy_bp.add_url_rule('/statistics/', 'statistics', RHCategoryStatistics)
+_legacy_bp.add_url_rule('/events.atom', 'categoryDisplay-atom', categoryDisplay.RHCategoryToAtom)
+_legacy_bp.add_url_rule('/events.rss', 'categoryDisplay-rss',
+                 make_compat_redirect_func(_legacy_bp, 'categoryDisplay-atom'))
+_legacy_bp.add_url_rule('/events.ics', 'categoryDisplay-ical', categoryDisplay.RHCategoryToiCal)
+_legacy_bp.add_url_rule('/icon', 'categoryDisplay-getIcon', categoryDisplay.RHCategoryGetIcon)
+
+# Overview
+_legacy_bp.add_url_rule('/overview', 'categOverview', categoryDisplay.RHCategOverviewDisplay)
+_legacy_bp.add_url_rule('!/category/<selCateg>/overview', 'categOverview', categoryDisplay.RHCategOverviewDisplay)
+_legacy_bp.add_url_rule('/overview', 'categOverview', categoryDisplay.RHCategOverviewDisplay)
+_legacy_bp.add_url_rule('/overview.rss', 'categOverview-rss',
+                 make_compat_redirect_func(_legacy_bp, 'categoryDisplay-atom'))
+
+# Event map
+_legacy_bp.add_url_rule('/map', 'categoryMap', categoryDisplay.RHCategoryMap)
+
+# Event calendar
+_legacy_bp.add_url_rule('!/category/calendar/', 'wcalendar', calendar.RHCalendar)
+_legacy_bp.add_url_rule('!/category/calendar/select', 'wcalendar-select', calendar.RHCalendarSelectCategories,
+                 methods=('GET', 'POST'))
+
+_bp = IndicoBlueprint('categories', __name__, template_folder='templates', virtual_template_folder='categories',
```

Comment:
move this above the legacy one (maybe just amend @pferreir's commit if you didn't change anything else in here)